### DISCUSSION TITLE
Call an secured API from postgres

### DISCUSSION THREAD

**Discussion by psteinroe** [supabase/supabase#4913] [Questions]
Hi,

I am looking into how to call an external API from within a Postgres trigger using the provided http extension and wondering how to best pass an API secret.

It seems like there is neither a lot of documentation nor many discussions around this topic:
- [This video](https://www.youtube.com/watch?v=rARgrELRCwY) only makes un-authenticated requests
- [In this approach](https://github.com/moein/supabase-migration) the secret is set into the sql file during the migration

Both solution do not seem optimal to me. Other ideas are
- Writing the secret into the database using a crypto extension
- Passing the JWT token from the request

For our use cases, passing the JWT token would be the simplest way of handling auth. We could check the token within the function and authenticate the request with it. Hence, It would be great to access the JWT similar to `auth.uid()` retrieves the id of the user. 

What is the preferred solution here? And are there already plans for this on the roadmap? 

Thanks!

PS: Love Supabase so far! 

**EDIT:** I will test it tomorrow, but [it seems like](https://postgrest.org/en/stable/api.html#accessing-request-headers-cookies-and-jwt-claims) it is possible to retrieve the JWT from the header using `SELECT current_setting('request.headers', true)::json->>'Authorization';`

**EDIT 2:** It works as expected (using the pgsql-http extension instead of pg_net because of #4969)
 ```sql
 perform *
  from
    http((
          'POST',
          v_send_message_webhook_url,
          ARRAY[http_header('Authorization',(SELECT current_setting('request.headers', true)::json->>'authorization'))],
          'application/json',
          v_webhook_payload::text
        )::http_request);
```

**Comment by sarkarology** [comment]
This helped. Thanks!
